src/core/core/rpc_engine.cpp

```diff
@@ -846,14 +850,28 @@ namespace dsn {
                             {
                                 resolver->on_access_failure(req2->header->gpid.u.partition_index, err);
 
+                                /*
+                                // TODO(qinzuoyan): because the user will fetch reply states from this response_task 'call',
```

Comment:
One possible way may be to reset the task state to ```ready``` from ```running``` instead of ```finished```, so that the task can be triggered again. I guess you may have a try?